_posts/2020-06-02-ros-2.md

```diff
@@ -0,0 +1,90 @@
+ ```
+
+### Setup your sources list
+
+Now that we are able to install software from all sources, we would like to make our target system be able to accept software from [ros.org](http://www.ros.org). Open up a terminal, and run the following command:
+
+```shell
+sudo sh -c 'echo "deb http://packages.ros.org/ros2/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2-latest.list'
+```
+What you are doing here is adding the Debian package for ROS on your `sources.list.d` directory so that the package is able to add new sources without changing the rest of the configuration.
+
+### Install
+
+Now we are finally ready to begin installation. Update your package index first by running the following command:
+```shell
+sudo apt update
+```
+
+Once the package list has been updated, you can now proceed to install ROS2 Dashing. Here, you have a choice to make, largely based on what you are trying to achieve with your target system when you will use ROS2. If you can afford the memory, it is honestly the best to always go for the **Desktop** install, as this has all the functionalities ROS2 will have to offer to you. But mind you, this is a fairly large(~2GB) file. The differences in the choices are highlighted below:
+
+* **Desktop:** Always recommended that you install this if you are a beginner, as this is basically ROS with all its features like multiple simulators, robot libraries, perception support and a whole lot. This is the best choice if you want to start development the moment after you finish installing ROS.
+
+* **ROS-Base:** The lightest version of ROS out there. This is generally meant for edge devices, and contains the core ROS functionalities, the ROS build tools, and communication libraries. Ideal for extremely resource constrained devices. Keep in mind though, that with this version you will not get any form of GUI.
+
+Once you have made an informed decision based on your requirements from ROS, you can install the packages by typing the command on your terminal:
+
+```shell
+sudo apt-get install ros-dashing-desktop
+```
+
+You will note in the previous command that we have installed the `desktop` version. If you want to install the other versions, replace the `desktop` in the previous command by `ros-base` such that the commands would look like this:
+
+```shell
+sudo apt-get install ros-dashing-ros-base
+```
+
+Congratulations! You have now successfully configured ROS2 on your target system. However, in order to start using the system, you would want to follow the next step, which does a little bit of book keeping.
+
+### Install dependencies
+
+
+
+### Setup the Environment
+
+With the completion of the previous step, ROS2 is now installed in our **/opt/ros/dashing** directory. Now, since we are using ROS2 and ROS side-by-side, we will source the ROS2 workspace only when we need it. We cannot put both the setup files in the .bashrc file, this would only source the last entry.
```

Comment:
Add the shell script line to source ROS2

Reply:
Did it in commit 882f8ec